Sources/Views/MusicProgressSliderCell.m

```diff
@@ -0,0 +1,55 @@
+//
+//  MusicProgressSlider.m
+//  Hermes
+//
+//  Created by Xinhong LIU on 19/4/15.
+//
+//
+
+#import "MusicProgressSliderCell.h"
+
+@implementation MusicProgressSliderCell
+
+@synthesize knobImage = _knobImage;
+
+- (NSImage *)knobImage {
+  return _knobImage;
+}
+
+- (void)setKnobImage:(NSImage *)knobImage {
+  _knobImage = knobImage;
+  if (_knobImage == nil) {
+    // default knob image
+    _knobImage = [NSImage imageNamed:@"knob.png"];
+  }
+}
+
+- (id)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self setKnobImage:[NSImage imageNamed:@"knob.png"]];
+  }
+  return self;
+}
+
+- (NSRect)knobRectFlipped:(BOOL)flipped {
+  float KNOB_WIDTH = [[self knobImage] size].width;
+  float KNOB_HEIGHT = [[self knobImage] size].height;
+  CGFloat percentage = ([self doubleValue]  - [self minValue])
+  / ([self maxValue] - [self minValue]);
+  CGSize sliderSize = [self barRectFlipped:false].size;
+  CGPoint sliderOrigin = [self barRectFlipped:false].origin;
+  NSRect knobRect = NSMakeRect(0, 0, 0, 0);
+  knobRect.size.width = KNOB_WIDTH;
+  knobRect.size.height = KNOB_HEIGHT;
+  knobRect.origin.x = (int) // int make it smooth
+  (sliderOrigin.x + percentage * (sliderSize.width - KNOB_WIDTH));
```

Comment:
Again, hanging indent. More importantly, if you are wanting to lose the fractional part of a float, and keep it a float, it might be clearer to use [`truncf(3)`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/truncf.3.html) from `math.h`.

`knobRect.origin.x = truncf(sliderOrigin.x + percentage*(sliderSize.width - knobWidth));` kind of thing